quicktest_test.go

```diff
@@ -0,0 +1,89 @@
+// Licensed under the MIT license, see LICENCE file for details.
+
+package quicktest_test
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+)
+
+// testingT can be passed to qt.New for testing purposes.
+type testingT struct {
+	*testing.T
+	errorBuf bytes.Buffer
+	fatalBuf bytes.Buffer
+}
+
+// Error overrides *testing.T.Error so that messages are collected.
+func (t *testingT) Error(a ...interface{}) {
+	fmt.Fprint(&t.errorBuf, a...)
+}
+
+// Fatal overrides *testing.T.Fatal so that messages are collected and the
+// goroutine is not killed.
+func (t *testingT) Fatal(a ...interface{}) {
+	fmt.Fprint(&t.fatalBuf, a...)
+}
+
+// errorString returns the error message.
+func (t *testingT) errorString() string {
+	return t.errorBuf.String()
+}
+
+// fatalString returns the fatal error message.
+func (t *testingT) fatalString() string {
+	return t.fatalBuf.String()
+}
+
+// assertPrefix fails if the got value does not have the given prefix.
+func assertPrefix(t testing.TB, got, prefix string) {
+	if h, ok := t.(helperer); ok {
+		h.Helper()
+	}
+	if prefix == "" {
+		t.Fatal("prefix: empty value provided")
+	}
+	if !strings.HasPrefix(got, prefix) {
+		t.Fatalf("prefix:\ngot  %q\nwant %q:\n-----------------------------------\n%s", got, prefix, got)
+	}
+}
+
+// assertErrHasPrefix fails if the given error is nil or does not have the
+// given prefix.
+func assertErrHasPrefix(t testing.TB, err error, prefix string) {
+	if h, ok := t.(helperer); ok {
+		h.Helper()
+	}
+	if err == nil {
+		t.Fatalf("error:\ngot  nil\nwant %q", prefix)
+	}
+	assertPrefix(t, err.Error(), prefix)
+}
+
+// assertErrIsNil fails if the given error is not nil.
+func assertErrIsNil(t testing.TB, err error) {
+	if h, ok := t.(helperer); ok {
+		h.Helper()
+	}
+	if err != nil {
+		t.Fatalf("error:\ngot  %q\nwant nil", err)
+	}
+}
+
+// assertBool fails if the given boolean values don't match.
+func assertBool(t testing.TB, got, want bool) {
+	if h, ok := t.(helperer); ok {
+		h.Helper()
+	}
+	if got != want {
+		t.Fatalf("bool:\ngot  %v\nwant %v", got, want)
+	}
+}
+
+// helperer is used to check whether the current Go version supports testing
+// helpers.
+type helperer interface {
```

Comment:
surely not! :) just "helper" should be OK, no?

Reply:
Definitively.